geode-core/src/main/java/org/apache/geode/internal/cache/DiskStoreImpl.java

```diff
@@ -236,16 +239,16 @@ public static boolean getBoolean(String sysProp, boolean def) {
    * by default be ON but in order to switch it off you need to explicitly
    */
   static final boolean PREALLOCATE_IF =
-      !System.getProperty(DistributionConfig.GEMFIRE_PREFIX + "preAllocateIF", "true")
-          .equalsIgnoreCase("false");
+      System.getProperty(DistributionConfig.GEMFIRE_PREFIX + "preAllocateIF", "true")
+          .equalsIgnoreCase("true");
```

Comment:
I changed this because I think its easy to read. I think the result of the function for correct values (that is "false" and "true") is the same than before, but now I see something I missed: in case a typo is introduced in the property value (i.e. "fals"), with the old code the default behaviour (preallocate = true) would be configured, and with the new code, the preallocate would be configured as false. I think its better to deactivate the default behaviour only when "false" is written. Otherwise, the preallocate could be deactivated if the value of the property is mispelled "tru", for example. Thanks for this comment!